### PR TITLE
South African Setswana Language

### PR DESCRIPTION
**Pull request by Shumie82**
## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [ ] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by Shumie82** [comment]
Please approve the additional of South African Setswana language to your repository. Thanks

File: spacy/lang/tn/__init__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from .stop_words import STOP_WORDS
+from .lex_attrs import LEX_ATTRS
+from .punctuation import TOKENIZER_INFIXES
+from ...language import Language
+
+
+class SetswanaDefaults(Language.Defaults):
+    suffixes = TOKENIZER_SUFFIXES
```
Comment (svlandeg) [inline spacy/lang/tn/__init__.py:8]: I think you want to do this instead?
```suggestion
    infixes = TOKENIZER_INFIXES
```

File: spacy/lang/tn/examples.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# coding: utf8
+from __future__ import unicode_literals
+
+
```
Comment (svlandeg) [inline spacy/lang/tn/examples.py:4]: You can just remove these lines - they were only relevant when we were still supporting Python 2
```suggestion
```

File: spacy/lang/tn/lex_attrs.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
+coding: utf8
+
+from __future__ import unicode_literals
+
```
Comment (svlandeg) [inline spacy/lang/tn/lex_attrs.py:4]: You can just remove these lines - they were only relevant when we were still supporting Python 2
```suggestion
```

File: spacy/lang/tn/stop_words.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+coding: utf8
+
+from __future__ import unicode_literals
+
+
```
Comment (svlandeg) [inline spacy/lang/tn/stop_words.py:5]: You can just remove these lines - they were only relevant when we were still supporting Python 2
```suggestion
```

File: spacy/lang/tn/stop_words.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+eo yona tseraganyo eng ne sentle re rona thata 
+godimo fitlha pedi masomamabedi lesomepedi mmogo 
+tharo tseo boraro tseno yone jaanong bobona bona 
+lesome tsaya tsamaiso nngwe masomethataro thataro 
+tsa mmatota tota sale thoko supa dira tshwanetse di mmalwa masisi
+bonala e tshwanang bogolo tsenya tsweetswee karolo 
+sepe tlhalosa dirwa robedi robongwe lesomenngwe gaisa 
+tlhano lesometlhano botlalo lekgolo           
+""".split())
+print(STOP_WORDS)
```
Comment (svlandeg) [inline spacy/lang/tn/stop_words.py:24]:
```suggestion
```

File: spacy/lang/tn/examples.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# coding: utf8
+from __future__ import unicode_literals
+
+
+"""
+Example sentences to test spaCy and its language models.
+>>> from spacy.lang.en.examples import sentences
```
Comment (svlandeg) [inline spacy/lang/tn/examples.py:7]:
```suggestion
>>> from spacy.lang.tn.examples import sentences

```

**Review by svlandeg — CHANGES_REQUESTED** [review]
Thanks for the PR! It would definitely be good to have support for Setswana in spaCy!

I've added a few detailed comments and suggested changes. You should be able to just accept these changes on the Github interface (if you agree with them, ofcourse).

One other question I have is whether you could include a few unit tests that showcase that the processing of a Setswana text goes ok. The tests are stored at `spacy/tests/lang`. https://github.com/explosion/spaCy/blob/master/spacy/tests/lang/ti/test_text.py is a good example: a few tests to check that a sentence is segmented in the correct number of words, and a test that checks whether the `like_num` property works as expected for a variety of Setswana words. It would be great if you could add some tests like those!

**Comment by Shumie82** [comment]
Thanks Sofie for the comments and review.  Please review the changes I have made.
For the tests, I will submit another pull request as soon as I am done. Just to make sure I have checked all that needs to be done properly. 
Thanks.

**Comment by ines** [comment]
Thanks so much for your work on this 👍 I made one small update and all tests are green now. I'll go ahead and merge this PR. You can then submit another pull request with some tests whenever you're ready.

**Review by ines — APPROVED** [review]

**Pull request merged**